tests/integrations/raftstore/test_transfer_leader.rs

```diff
@@ -110,12 +110,73 @@ fn test_pd_transfer_leader<T: Simulator>(cluster: &mut Cluster<T>) {
     }
 }
 
+fn test_pd_transfer_leader_multi_target<T: Simulator>(cluster: &mut Cluster<T>) {
+    let pd_client = Arc::clone(&cluster.pd_client);
+    pd_client.disable_default_operator();
+
+    cluster.run();
+
+    cluster.must_put(b"k1", b"v1");
+    cluster.add_send_filter(IsolationFilterFactory::new(3));
+    cluster.must_put(b"k2", b"v2");
+    // append index of the cluster will finally be 2, 2, 1
+
+    // call command on this leader directly, must successfully.
+    let mut region = cluster.get_region(b"");
+    let mut req = new_request(
+        region.get_id(),
+        region.take_region_epoch(),
+        vec![new_get_cmd(b"k1")],
+        false,
+    );
+
+    // select multi target leaders to transfer
+    // set `peer` to 3, make sure the new leader comes from `peers`
+    pd_client.transfer_leader(
+        region.get_id(),
+        new_peer(3, 3),
+        vec![new_peer(2, 2), new_peer(3, 3)],
+    );
+
+    for _ in 0..100 {
+        // reset leader and wait transfer successfully.
+        cluster.reset_leader_of_region(1);
+
+        sleep_ms(20);
+
+        if let Some(leader) = cluster.leader_of_region(1) {
+            if leader.get_id() == 2 {
+                // make sure new leader apply an entry on its term
+                // so we can use its local reader safely
+                cluster.must_put(b"k2", b"v2");
```

Comment:
It is unnecessary, the newly elected leader will append a noop entry on its term, and only serve read requests after the noop entry is applied.